File: backend/prisma/smoke-auth.ts
import { database } from "../src/database/client.js";
import { AuthService } from "../src/modules/auth/auth.service.js";
import type { MailService } from "../src/modules/notifications/mail.service.js";

const email = `auth-smoke-${Date.now()}@example.invalid`;
const password = "SmokeTestPassword123";
let verificationCode: string | undefined;

const mail: MailService = {
  async sendEmailVerification(_email, code) {
    verificationCode = code;
  },
  async sendPasswordReset() {},
  async sendInvitation() {},
};

async function main(): Promise<void> {
  const service = new AuthService(database, mail);
  let completed = false;
  try {
    await service.register({
      email,
      password,
      firstName: "Auth",
      lastName: "Smoke",
    });
    if (!verificationCode) throw new Error("Registration did not produce a verification code");

    await service.verifyEmail({ email, code: verificationCode });
    const result = await service.login(
      { email, password },
      { ipAddress: "127.0.0.1", userAgent: "esquare-auth-smoke-test" },
    );
    if (!result.accessToken || !result.refreshToken) {
      throw new Error("Login did not produce both session tokens");
    }
    console.log("Auth smoke test passed: register, verify, and login all succeeded.");
    completed = true;
  } finally {
    try {
      await database.user.deleteMany({ where: { email } });
    } catch (error) {
      if (completed) throw error;
      console.error("Auth smoke cleanup also failed:", error);
    }
    await database.$disconnect();
  }
}

void main();

File: backend/scripts/smoke_auth.py
from __future__ import annotations

from time import time_ns

from fastapi.testclient import TestClient
from sqlalchemy import text

import app.auth as auth_module
from app.database import engine
from app.main import app


def main() -> None:
    email = f"auth-smoke-{time_ns()}@example.com"
    password = "SmokeTestPassword123"
    codes: list[str] = []
    original = auth_module.send_verification
    auth_module.send_verification = lambda _email, code: codes.append(code)
    client = TestClient(app)
    try:
        registered = client.post(
            "/api/v1/auth/register",
            json={
                "email": email,
                "password": password,
                "firstName": "Auth",
                "lastName": "Smoke",
            },
        )
        assert registered.status_code == 202 and codes
        verified = client.post("/api/v1/auth/verify-email", json={"email": email, "code": codes[0]})
        assert verified.status_code == 200
        logged_in = client.post("/api/v1/auth/login", json={"email": email, "password": password})
        assert logged_in.status_code == 200 and logged_in.json().get("accessToken")
        print("Auth smoke test passed: register, verify, and login all succeeded.")
    finally:
        auth_module.send_verification = original
        with engine.begin() as connection:
            connection.execute(text("DELETE FROM users WHERE email=:email"), {"email": email})


if __name__ == "__main__":
    main()
